### PR TITLE
Index page fetch

### PR DESCRIPTION
**Pull request by arpepper139**
We updated our index page to use fetch, sorted into FloofCategoryTiles based on category, did some styling, added a header to our FloofCategoryTile and updated our tests accordingly. All tests were passing.

### CODE REVIEW

File: app/javascript/components/FloofCategoryTile.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ const FloofCategoryTile = (props) => {
 
   return(
     <div className='floof_cat_tile'>
+      <header>{ props.floofData[0].category }</header>
```
Comment (dpickett) [inline app/javascript/components/FloofCategoryTile.js:20]: what happens if there is not a floof for the given category?

File: app/javascript/containers/FloofIndexContainer.js [hunk 2]
```diff
@@ -5,28 +5,59 @@ class FloofIndexContainer extends Component {
...
-        job_title: 'Junior Associate',
-        picture: 'http://teachyourpetstotype.com/dogtyping4.jpg'
-      }
-    ]
+      floofs: []
     }
   }
 
+  componentDidMount() {
+    fetch('/api/v1/floofs')
```
Comment (dpickett) [inline app/javascript/containers/FloofIndexContainer.js]: we should be explicit here in fetching `/api/v1/floofs.json`

File: app/javascript/containers/FloofIndexContainer.js [hunk 3]
```diff
@@ -5,28 +5,59 @@ class FloofIndexContainer extends Component {
...
+
+    if (floofData !== []) {
+      //create a list of categories
+      let categories = floofData.map((floof) => {
+        return floof.category
+      })
+      categories = categories.filter((category, index, categories) => categories.indexOf(category) === index);
+      categories = categories.sort()
+      
+      //map through our list of categories
```
Comment (dpickett) [inline app/javascript/containers/FloofIndexContainer.js]: this processing and work should be done on the API side - you can leverage the power of the database to help with this.

**Review by dpickett — DISMISSED** [review]

File: app/javascript/components/FloofCategoryTile.js (hunk 1 above)
Comment (arpepper139) [reply app/javascript/components/FloofCategoryTile.js:20]: I think based on the way the logic is currently set up a FloofCategoryTile will only be created if a floof is found with that category and that floof's category is then added to the categories array in FloofIndexContainer, which means there should always be at least one floof in a category tile. Is there anything I'm missing there?

**Pull request merged**